moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -330,255 +116,185 @@ export function useMapDrawTools(
   const featureTypesEqual = (featureA, featureB) =>
     (featureA?.geometry?.type ?? 1) === (featureB?.geometry?.type ?? 0);
 
-  /**
-   * Add existing user drawn points in the project extent feature collection to the draw UI so they are editable
-   */
-  const initializeExistingDrawFeatures = useCallback(
-    ref => {
-      if (ref) {
-        // Only add features that are not already present in the draw UI to avoid duplicates
-        const drawnFeatures = getDrawnFeaturesFromFeatureCollection(
-          featureCollection
-        );
+  const initializeExistingDrawFeatures = () => {
+    // Only add features that are not already present in the draw UI to avoid duplicates
+    const drawnFeatures =
+      getDrawnFeaturesFromFeatureCollection(featureCollection);
 
-        // Collect all the features in the map
-        // getFeatures() is a react-map-gl-draw function
-        const featuresAlreadyInDrawMap = ref.getFeatures();
+    // Collect all the features that the draw UI is tracking
+    const featureCollectionAlreadyInDrawMap = drawControlsRef.current.getAll();
+    const featuresAlreadyInDrawMap = featureCollectionAlreadyInDrawMap.features;
 
-        // Retrieve only the features that are present in state, but not the map
-        const featuresToAdd = findDifferenceByFeatureProperty(
-          "PROJECT_EXTENT_ID",
-          drawnFeatures,
-          featuresAlreadyInDrawMap
-        );
+    // Retrieve only the features that are present in state, but not the map
+    const featuresToAdd = findDifferenceByFeatureProperty(
+      "PROJECT_EXTENT_ID",
+      drawnFeatures,
+      featuresAlreadyInDrawMap
+    );
 
-        // Draw the features
-        ref.addFeatures(featuresToAdd);
-      }
-    },
-    [featureCollection]
-  );
+    // Draw the features
+    featuresToAdd.forEach((feature) => drawControlsRef.current.add(feature));
+  };
 
   /**
    * Updates state and mutates additions and deletions of points drawn with the UI
    */
-  const saveDrawnPoints = (runActionDispatch = true) => {
-    // If there is a map reference, get it's features or assume empty array
-    const drawnFeatures = mapEditorRef.current
-      ? mapEditorRef.current.getFeatures()
-      : [];
```

Comment:
`saveDrawnPoints` is now called through a create event handler which gives us the new feature data so we don't need to query the draw tools anymore